packages/styled-base/types/tests.tsx

```diff
@@ -116,13 +116,15 @@ const Button3 = styled<'button', PrimaryProps>('button')(props => ({
 // $ExpectError
 ;<Button3 type="button" />
 
-const Button4 = styled<ReactClassProps0, PrimaryProps>(ReactClassComponent0)`
+const Button4 = styled<typeof ReactClassComponent0, PrimaryProps>(
+  ReactClassComponent0
+)`
```

Comment:
Why does there have to be `typeof ReactClassComponent0` if the user is passing `ReactClassComponent0` to styled, can't TS infer it?

Reply:
@mitchellhamilton Since we need `PrimaryProps`. It's same with
https://github.com/emotion-js/emotion/blob/952a2e442c6528a615f5719c5896a14c11deebfa/packages/styled-base/types/tests.tsx#L85-L98

Reply:
In TS3.2, partial type argument will be supported (though it could be deferred...) and it will become much easier to use these.